tools/codemods/src/codemods/popover-v12/transform.ts

```diff
@@ -25,85 +62,266 @@ const propNamesToRemove = [
- * 6. Removes `shouldTooltipUsePortal` prop from `Copyable` component
+ * Transformer function that will transform the below packages by default. Consumers can
+ * use the `--packages` flag to apply the codemod to a subset of these packages.
+ *
+ * It does the following:
+ * 1. Adds an explicit `usePortal={true}` declaration if left undefined and consolidates
+ *    the `usePortal` and `renderMode` props into a single `renderMode` prop for components
+ *    in the following packages:
+ * - `@leafygreen-ui/combobox`
+ * - `@leafygreen-ui/menu`
+ * - `@leafygreen-ui/popover`
+ * - `@leafygreen-ui/select`
+ * - `@leafygreen-ui/split-button`
+ * - `@leafygreen-ui/tooltip`
+ *
+ * 2. Removes `popoverZIndex`, `portalClassName`, `portalContainer`, `portalRef`,
+ *    `scrollContainer`, and `usePortal` props from the components in the following packages:
+ * - `@leafygreen-ui/info-sprinkle`
+ * - `@leafygreen-ui/inline-definition`
+ * - `@leafygreen-ui/number-input`
+ *
+ * 3. Removes `popoverZIndex`, `portalClassName`, `portalContainer`, `portalRef`, and
+ *    `scrollContainer` props from components in the following packages:
+ * - `@leafygreen-ui/date-picker`
+ * - `@leafygreen-ui/guide-cue`
+ *
+ * 4. Removes `popoverZIndex`, `portalClassName`, `portalContainer`, `scrollContainer`,
+ *    and `usePortal` props from `Code` component in the `@leafygreen-ui/code` package
+ *
+ * 5. Removes `portalClassName`, `portalContainer`, `portalRef`, `scrollContainer`, and
+ *    `usePortal` props from `SearchInput` component in the `@leafygreen-ui/search-input` package
+ *
+ * 6. Removes `shouldTooltipUsePortal` prop from the `Copyable` component in the `@leafygreen-ui/copyable` package
+ *
+ * 7. Replaces `justify="fit"` prop value with `justify="middle"` for components in the following packages:
+ * - `@leafygreen-ui/date-picker`
+ * - `@leafygreen-ui/info-sprinkle`
+ * - `@leafygreen-ui/inline-definition`
+ * - `@leafygreen-ui/menu`
+ * - `@leafygreen-ui/popover`
+ * - `@leafygreen-ui/tooltip`
  *
  * @param file the file to transform
  * @param jscodeshiftOptions an object containing at least a reference to the jscodeshift library
  * @param options an object containing options to pass to the transform function
  * @returns Either the modified file or the original file
  */
-export default function transformer(file: FileInfo, { jscodeshift: j }: API) {
+export default function transformer(
+  file: FileInfo,
+  { jscodeshift: j }: API,
+  options: MigrateOptions & Options,
+): string {
   const source = j(file.source);
+  const providedPackages = options.packages;
+
+  /**
+   * If the `packages` option is provided, ensure that the provided packages are all valid.
+   */
+  if (providedPackages) {
+    providedPackages.forEach(packageName => {
+      if (!defaultPackages.includes(packageName)) {
+        throw new Error(
+          `Cannot run popover-v12 codemod on package: ${packageName}`,
+        );
```

Comment:
Do we want to `Throw`, or just log an error and remove it? Don't have a strong opinion, just a thought